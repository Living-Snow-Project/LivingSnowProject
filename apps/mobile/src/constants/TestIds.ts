export const TestIds = {
  GPS: {
    GpsManualEntryModal: "gps-enable-manual-coordinates-modal",
    GpsManualEntryPressable: "gps-manual-pressable",
  },
  Selectors: {
    AlgaeColor: "algae-color-selector",
    AlgaeSize: "algae-size-selector",
    Photos: "photo-selector",
    RecordType: "record-type-selector",
  },
  RecordScreen: {
    UploadButton: "record-screen-upload-button",
    UpdateButton: "record-screen-update-button",
  },
  TimelineScreen: {
    FlatList: "timeline-screen-flatlist",
    RecordListView: "timeline-screen-record-list-view",
    NewRecordButton: "timeline-screen-record-button",
    SettingsButton: "timeline-screen-settings-button",
    ScrollToTopButton: "timeline-screen-scroll-to-top-button",
  },
  SettingsScreen: {
    ShowGpsWarning: "settings-screen-show-gps-warning",
  },
  Icons: {
    AddPhotosIcon: "add-photos-icon",
    DeleteIcon: "delete-icon",
    DoneSelectingPhotosIcon: "done-selecting-photos-icon",
    EditIcon: "edit-icon",
    PictureIcon: "picture-icon",
    SampleIcon: "sample-icon",
    SaveIcon: "save-icon",
    ScrollTopIcon: "scroll-top-icon",
    SightingIcon: "sighting-icon",
    SnowIcon: "snow-icon",
  },
  RecordList: {
    MenuTrigger: "pending-record-menu-trigger",
    EditRecordAction: "edit-record-action",
    DeleteRecordAction: "delete-record-action",
  },
  Modal: {
    ConfirmButton: "modal-confirm-button",
    DeletePendingRecord: "delete-pending-record",
    DiskUsageDeletePhotos: "disk-usage-delete-photos-modal",
    NoButton: "modal-no-button",
  },
};
