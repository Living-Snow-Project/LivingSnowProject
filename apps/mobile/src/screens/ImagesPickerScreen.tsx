import React from "react";
import { Platform } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { ImagesPickerScreenProps } from "../navigation/Routes";
/* eslint-disable import/no-relative-packages */
import { AssetsSelector } from "../../expo-images-picker";
import { SettingsType, StylesType } from "../../expo-images-picker/src/Types";

export default function ImagesPickerScreen({
  navigation,
  route,
}: ImagesPickerScreenProps) {
  const widgetSettings: SettingsType = {
    getImageMetaData: false,
    initialLoad: 100,
    assetsType: ["photo"],
    minSelection: 0,
    maxSelection: 4,
    portraitCols: 4,
    landscapeCols: 5,
  };

  const widgetStyles: StylesType = {
    margin: 3,
    bgColor: "white",
    spinnerColor: "blue",
    widgetWidth: 100,
    screenStyle: {
      borderRadius: 5,
      overflow: "hidden",
    },
    widgetStyle: {
      margin: 10,
    },
    videoIcon: {
      Component: Ionicons,
      iconName: Platform.OS === "ios" ? "ios-videocam" : "md-videocam",
      color: "white",
      size: 20,
    },
    selectedIcon: {
      Component: Ionicons,
      iconName:
        Platform.OS === "ios"
          ? "ios-checkmark-circle-outline"
          : "md-checkmark-circle-outline",
      color: "grey",
      bg: "lightgrey",
      size: 40,
    },
  };

  const widgetNavigator = {
    Texts: {
      finish: "Finish",
      back: "Back",
      selected: "",
    },
    buttonTextStyle: {},
    buttonStyle: {},
    // TODO: this should be on react-navigation header not the component's
    // TODO: possible to eliminate goBackText and backFunction?
    onBack: () => () => {},
    onSuccess: (data: any[]) => {
      route.params.onUpdatePhotos(data);
      navigation.goBack();
    },
  };

  const widgetResize = {
    compress: 0.4,
    width: 1024, // TODO: ideally this would be the size constraint on the largest axis
    base64: false,
    saveTo: "jpeg",
  };

  return (
    // TODO: how to tell what's already selected
    <AssetsSelector
      Settings={widgetSettings}
      Styles={widgetStyles}
      Errors={{}}
      Navigator={widgetNavigator}
      Resize={widgetResize}
    />
  );
}
