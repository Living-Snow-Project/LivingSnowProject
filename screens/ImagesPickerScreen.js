import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { AssetsSelector } from 'expo-images-picker';
import { Ionicons } from '@expo/vector-icons';

function ImagesPickerScreen(props) {
  let {navigation } = props;
  return (
    <AssetsSelector
      options={{
        manipulate: {
          compress: 0.7,
          base64: false,
          saveTo: 'jpeg',
        },
        assetsType: ['photo'],
        maxSelections: 3,
        margin: 3,
        portraitCols: 4,
        landscapeCols: 5,
        widgetWidth: 100,
        widgetBgColor: 'white',
        selectedBgColor: 'red',
        spinnerColor: 'blue',
        videoIcon: {
          Component: Ionicons,
          iconName: 'ios-videocam',
          color: 'white',
          size: 20,
        },
        selectedIcon: {
          Component: Ionicons,
          iconName: 'ios-checkmark-circle-outline',
          color: 'white',
          bg: 'white',
          size: 20,
        },
        defaultTopNavigator: {
          continueText: 'Finish',
          goBackText: 'Back',
          //buttonStyle: validViewStyleObject,
          //textStyle: validTextStyleObject,
          backFunction: navigation.goBack,
          doneFunction: (data) => navigation.navigate('Record', {data: data}),
        },
        noAssets: {
          Component: Text,
        },
      }}
    />
  );
}

ImagesPickerScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  route: PropTypes.object.isRequired,
}

export { ImagesPickerScreen };
