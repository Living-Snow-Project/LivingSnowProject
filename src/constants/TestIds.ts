const TestIds = {
  GPS: {
    gpsManualInputTestId: "gps-manual-input",
    gpsManualPressableTestId: "gps-manual-pressable",
  },
  Photos: {
    photoSelectorTestId: "photo-selector",
  },
  Pickers: {
    recordSelectorTestId: "record-type-picker",
    atlastSelectorTestId: "atlas-type-picker",
  },
  RecordScreen: {
    UploadButton: "record-screen-upload-button",
  },
  TimelineScreen: {
    RefreshControl: "timeline-screen-refresh-control",
    SettingsButton: "timeline-screen-settings-button",
    NewRecordButton: "timeline-screen-record-button",
  },
  SettingsScreen: {
    ShowGpsWarning: "settings-screen-show-gps-warning",
    ShowAtlasRecords: "settings-screen-show-atlas-records",
    ShowOnlyAtlasRecords: "settings-screen-show-only-atlas-records",
  },
  Icons: {
    SampleIcon: "sample-icon",
    SightingIcon: "sighting-icon",
    PictureIcon: "picture-icon",
    DeleteIcon: "delete-icon",
    EditIcon: "edit-icon",
    SnowIcon: "snow-icon",
  },
  RecordList: {
    EditRecordAction: "edit-record-action",
    DeleteRecordAction: "delete-record-action",
  },
};

export default TestIds;
