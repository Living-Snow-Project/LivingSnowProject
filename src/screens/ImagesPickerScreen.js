import React from 'react';
import { Platform, Text } from 'react-native';
import PropTypes from 'prop-types';
import { AssetsSelector } from '../../expo-images-picker/index';
import { Ionicons } from '@expo/vector-icons';

export const ImagesPickerScreen = ({navigation, route}) => {
  return (
    <AssetsSelector
      options={{
        manipulate: {
          compress: 0.4,
          width: 1024, // ideally this would be the size constraint on the largest axis
          base64: false,
          saveTo: 'jpeg',
        },
        assetsType: ['photo'],
        maxSelections: 4,
        margin: 3,
        portraitCols: 4,
        landscapeCols: 5,
        widgetWidth: 100,
        widgetBgColor: 'white',
        selectedBgColor: 'red',
        spinnerColor: 'blue',
        videoIcon: {
          Component: Ionicons,
          iconName: Platform.OS === 'ios' ? 'ios-videocam' : 'md-videocam',
          color: 'white',
          size: 20,
        },
        selectedIcon: {
          Component: Ionicons,
          iconName: Platform.OS === 'ios' ? 'ios-checkmark-circle-outline' : 'md-checkmark-circle-outline',
          color: 'grey',
          bg: 'lightgrey',
          size: 40,
        },
        defaultTopNavigator: {
          continueText: 'Finish',
          goBackText: null,
          //buttonStyle: {borderWidth:2, borderColor:'red'},
          textStyle: {fontSize: 15},
          backFunction: () => {},
          doneFunction: data => {
            route.params.onUpdatePhotos(data);
            navigation.goBack();
          },
        },
        noAssets: <Text>No Assets</Text>,
      }}
    />
  );
}

ImagesPickerScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  route: PropTypes.object.isRequired
}
