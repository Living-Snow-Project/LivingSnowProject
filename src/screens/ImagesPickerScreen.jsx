import React from "react";
import { Platform, Text } from "react-native";
import PropTypes from "prop-types";
import { Ionicons } from "@expo/vector-icons";
import { AssetsSelector } from "../../expo-images-picker/index";

export default function ImagesPickerScreen({ navigation, route }) {
  return (
    <AssetsSelector
      options={{
        manipulate: {
          compress: 0.4,
          width: 1024, // TODO: ideally this would be the size constraint on the largest axis
          base64: false,
          saveTo: "jpeg",
        },
        assetsType: ["photo"],
        maxSelections: 4,
        margin: 3,
        portraitCols: 4,
        landscapeCols: 5,
        widgetWidth: 100,
        widgetBgColor: "white",
        selectedBgColor: "red",
        spinnerColor: "blue",
        videoIcon: {
          Component: Ionicons,
          iconName: Platform.OS === "ios" ? "ios-videocam" : "md-videocam",
          color: "white",
          size: 20,
        },
        selectedIcon: {
          Component: Ionicons,
          iconName:
            Platform.OS === "ios"
              ? "ios-checkmark-circle-outline"
              : "md-checkmark-circle-outline",
          color: "grey",
          bg: "lightgrey",
          size: 40,
        },
        defaultTopNavigator: {
          // TODO: this should be on react-navigation header not the component's
          continueText: "Finish",
          goBackText: null,
          // buttonStyle: {borderWidth:2, borderColor:'red'},
          textStyle: { fontSize: 15 },
          backFunction: () => {},
          doneFunction: (data) => {
            route.params.onUpdatePhotos(data);
            navigation.goBack();
          },
        },
        noAssets: <Text>No Assets</Text>,
      }}
    />
  );
}

ImagesPickerScreen.propTypes = {
  route: PropTypes.shape({
    params: PropTypes.shape({
      onUpdatePhotos: PropTypes.func.isRequired,
    }).isRequired,
  }).isRequired,
};
