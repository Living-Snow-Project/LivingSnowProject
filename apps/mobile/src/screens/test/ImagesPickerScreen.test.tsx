import React from "react";
import { Platform } from "react-native";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import {
  ImagesPickerScreenNavigationProp,
  ImagesPickerScreenRouteProp,
} from "../../navigation/Routes";
import ImagesPickerScreen from "../ImagesPickerScreen";
import TestIds from "../../constants/TestIds";

// record action button renders independently of the screen
let photoSelectedActionButton: () => JSX.Element;

const navigation = {} as ImagesPickerScreenNavigationProp;
navigation.navigate = jest.fn();
navigation.goBack = jest.fn();
navigation.setOptions = ({
  headerRight,
}: {
  headerRight: () => JSX.Element;
}) => {
  photoSelectedActionButton = headerRight;
};

const route = {
  params: {
    onUpdatePhotos: jest.fn(),
  },
} as unknown as ImagesPickerScreenRouteProp;

describe("ImagesPickerScreen test suite", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders ios", async () => {
    const { toJSON } = render(
      <ImagesPickerScreen navigation={navigation} route={route} />
    );

    const { getByTestId } = render(photoSelectedActionButton());

    expect(toJSON()).toMatchSnapshot();
    fireEvent.press(getByTestId(TestIds.Icons.DoneSelectingPhotosIcon));
    await waitFor(() => expect(navigation.goBack).toBeCalled());
    expect(route.params.onUpdatePhotos).toBeCalled();
  });

  test("renders android", async () => {
    Platform.OS = "android";

    render(<ImagesPickerScreen navigation={navigation} route={route} />);

    const { getByTestId } = render(photoSelectedActionButton());

    fireEvent.press(getByTestId(TestIds.Icons.DoneSelectingPhotosIcon));
    await waitFor(() => expect(navigation.goBack).toBeCalled());
    expect(route.params.onUpdatePhotos).toBeCalled();
    Platform.OS = "ios";
  });
});
