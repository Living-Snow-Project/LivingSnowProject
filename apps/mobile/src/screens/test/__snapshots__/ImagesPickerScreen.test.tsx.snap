// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesPickerScreen test suite renders ios 1`] = `
<View
  bgColor="white"
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "borderRadius": 5,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "marginBottom": 0,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 0,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "98%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 38,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={Object {}}
      >
        Back
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      0
       
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 38,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={Object {}}
      >
        Finish
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        },
      ]
    }
  >
    <ActivityIndicator
      color="blue"
      size="large"
    />
  </View>
</View>
`;
